lib/src/clojure_lsp/feature/add_missing_libspec.clj

```diff
@@ -269,32 +279,35 @@
                       missing-requires))))))
 
 (defn find-require-suggestions [zloc missing-requires db]
-  (when-let [sexpr (z/sexpr zloc)]
-    (if-let [alias-str (namespace sexpr)]
+  (when-let [sym (safe-sym zloc)]
+    (if-let [alias-str (namespace sym)]
       (find-alias-require-suggestions alias-str missing-requires db)
-      (find-refer-require-suggestions sexpr missing-requires db))))
+      (find-refer-require-suggestions sym missing-requires db))))
 
 (defn add-require-suggestion [zloc chosen-ns chosen-alias chosen-refer db]
```

Comment:
Not sure I understood the changes of this function on this PR, why do we need this? Could you elaborate a little more on the issue we are solving here?

Reply:
@mainej

Reply:
Yeah, my apologies... This should have changed the whitespace handling only, but it turned into more. I can revert some of it, if you'd like.

I think that I was confused about how the functions in this namespace work. There's a group that are dedicated to finding potential requires and imports (the code actions). And there's a group that inserts requires and imports into the ns (the commands). 

I _thought_ that the commands just took whatever the user gave them and inserted that into the ns (except for the "missing" commands, which magically figure out the correct require/import.) But that's not how most of them work. Instead, they take the user's input, then _check_ whether that input would have been suggested as a code action, and _then_ insert that into the namespace.

I can revert to that behavior, if you'd like. But, I still think that my initial impressions is valid. Shouldn't the commands do whatever the user says, even if that's not something that would have been suggested by the actions? That would let the user construct an invalid require/import, but perhaps they have a reason for that.

In any case, that should have been discussed in a separate Issue or PR. Sorry about that. In the meantime, the code allows the user to make more changes than they would have been able to before. But, since you can't figure that out via the actions, most users will never notice.

Anyway, please let me know whether you'd like me to revert this, or revert it and create a new PR, or whatever you think is best.